Start refactoring x.509 profile (#336)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

The Issuer identity MUST be bound to the Signed Statement by including an identifier in the protected header.
If the protected header includes multiple identifiers, all those that are registered by the Transparency Service MUST be checked.

> When using X.509 Signed Statements, the Transparency Service MUST build and validate a complete certification path from an Issuer's certificate to one of the root certificates most recently registered as a trust anchor by the Transparency Service.

Reviewer: Why do we most recently?

Reviewer 2: I think that should be "currently"

> When using X.509 Signed Statements, the Transparency Service MUST build and validate a complete certification path from an Issuer's certificate to one of the root certificates most recently registered as a trust anchor by the Transparency Service.

Reviewer 2: ```suggestion
When using X.509 Signed Statements, the Transparency Service MUST build and validate a complete certification path from an Issuer's certificate to one of the root certificates currently registered as a trust anchor by the Transparency Service.
```

Multiple Issuers can make the same Statement about a single Artifact, affirming multiple Issuers agree.

> At least one identifier representing one credential MUST be included in the protected header of the COSE Envelope.
> 
> For example, when x.509 is utilized, either `x5t`, `x5chain` or `kid` must be included in the protected header.

Reviewer 3: ```suggestion
```

Author: Why the deletion? Is this a deferral to the COSE spec?

Reviewer 3: https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/336#discussion_r1941469555

Additionally, `x5chain` that corresponds to either `x5t` or `kid` identifying the leaf certificate in the included certification path MAY be included in the unprotected header of the COSE Envelope.

> - When using x.509 certificates, support for either `x5t` or `x5chain` in the protected header is REQUIRED to implement.

Reviewer 3: ```suggestion
- When using x.509 certificates, support for either `x5t` or `x5chain` in the protected header is REQUIRED to implement.
```

^ This is all that is needed.
